test/DynamoCoreTests/Performance/PerformanceTests.cs

```diff
@@ -40,23 +53,83 @@ public object[] FindWorkspaces()
         [TestFixtureSetUp]
         public void SetupPerformanceTests()
         {
-            executionData = new List<(string, TimeSpan, TimeSpan, TimeSpan, TimeSpan)>();
+            executionData = new List<(string, TimeSpan, TimeSpan, TimeSpan, TimeSpan, TimeSpan)>();
+            memoryData =
+                new List<(string, (MemoryInfo, MemoryInfo), (MemoryInfo, MemoryInfo), (MemoryInfo, MemoryInfo))>();
+
         }
 
         [TestFixtureTearDown]
         public void TeardownPerformanceTests()
         {
-            Console.WriteLine("{0,50}{1,9}{2,9}{3,11}{4,9}{5,9}{6,11}", "Graph", "Old C", "Old E", "Old C+E", "New C", "New E", "New C+E");
-            executionData.ForEach(item =>
+            if (memoryData.Any())
             {
-                var (graph, oldEngineCompileTime, oldEngineExecutionTime, newEngineCompileTime, newEngineExecutionTime) = item;
-                Console.WriteLine("{0,50}{1,9:0.0}{2,9:0.0}{3,11:0.0}{4,9:0.0}{5,9:0.0}{6,11:0.0}", graph,
-                    oldEngineCompileTime.TotalMilliseconds, oldEngineExecutionTime.TotalMilliseconds,
-                    oldEngineCompileTime.TotalMilliseconds + oldEngineExecutionTime.TotalMilliseconds,
-                    newEngineCompileTime.TotalMilliseconds, newEngineExecutionTime.TotalMilliseconds,
-                    newEngineCompileTime.TotalMilliseconds + newEngineExecutionTime.TotalMilliseconds);
-            });
+                Console.WriteLine("{0,50}{1,9}{2,9}{3,9}{4,9}{5,9}{6,9}{7,9}{8,9}{9,9}{10,9}{11,9}{12,9}", "Graph",
```

Comment:
could we add a comment explaining what each of these codes mean? My eyes glaze over a bit staring at this 😉

Reply:
Will do.